static/js/popovers.js

```diff
@@ -328,7 +328,7 @@ exports.hide_user_profile = function () {
 exports.show_user_profile = function (user) {
     exports.hide_all();
 
-    const dateFormat = moment.localeData().longDateFormat("LL");
+    const dateFormat = "MMMM d, yyyy";
```

Comment:
Oh fine , Looking into it . 
Seems like <a href = 'https://date-fns.org/docs/I18n'>this</a> can get the job done .

Reply:
Hey , I currently don't see an alternative for <code>moment.localeData().longDateFormat("LL");</code> in <code>date-fns</code> to get locale format , However this is a function <code><a href = 'https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString'>toLocaleDateString()</a></code> which might get the job done .

Reply:
I think we want `date-fns`’ [`"PPP"`](https://date-fns.org/v2.16.1/docs/format).

Reply:
Actually `date-fns` requires you to pass the desired locale.

 `new Intl.DateTimeFormat("default", {dateStyle: "long"}).format(…)` seems to be the right thing ([`Intl.DateTimeFormat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat), [constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat)).

Reply:
Yeah , So <code>dateFormat</code> variable has to be eradicated from <a href ='https://github.com/zulip/zulip/blob/master/static/js/popovers.js'>popovers.js</a> , since it will no longer contain the format of date (in form of <code>"MMMM d, yyyy"</code> or other)

Reply:
It’s…not a big deal to “eradicate” a variable that’s mentioned five times. :slightly_smiling_face: You should make whatever changes are needed for the code to make sense.

In this case it probably makes sense to put the `Intl.DateTimeFormat` object in a variable, and `dateFormat` seems like a fine name even though it’s a different type than before.

```js
const dateFormat = new Intl.DateTimeFormat("default", {dateStyle: "long"});
…
dateFormat.format(parseISO(user.date_joined))
```

`get_custom_profile_field_data` needs a similar update too then.